### PR TITLE
chore: add some content metrics and make sure getDeployments won't fail

### PR DESCRIPTION
**Pull request by menduz**

### CODE REVIEW

File: content/src/ContentMetrics.ts [hunk 1]
```diff
@@ -25,3 +25,39 @@ export const REPOSITORY_QUEUE_EXECUTED_QUERIES = new Histogram({
   help: 'Time spent in seconds since the queries were added to the queue until they got resolved',
   labelNames: ['priority']
 })
+
+export const DCL_CONTENT_GARBAGE_COLLECTION_ITEMS_TOTAL = new Counter({
```
Comment (agusaldasoro) [inline content/src/ContentMetrics.ts:29]: Love this!

File: content/src/service/ServiceImpl.ts [hunk 2]
```diff
@@ -363,16 +363,21 @@ export class ServiceImpl implements MetaverseContentService, ClusterDeploymentsS
...
-      db.taskIf((task) =>
-        this.deploymentManager.getDeployments(task.deployments, task.content, task.migrationData, options)
-      )
+    return this.repository.reuseIfPresent(
+      task,
+      (db) =>
+        db.taskIf((task) =>
+          this.deploymentManager.getDeployments(task.deployments, task.content, task.migrationData, options)
+        ),
+      { priority: DB_REQUEST_PRIORITY.HIGH }
```
Comment (agusaldasoro) [inline content/src/service/ServiceImpl.ts:372]: I'm not sure that this change should be together with the metrics. Also, are there any requests that are of LOW priority with this change?

File: content/src/ContentMetrics.ts [hunk 3]
```diff
@@ -25,3 +25,39 @@ export const REPOSITORY_QUEUE_EXECUTED_QUERIES = new Histogram({
...
+  labelNames: []
+})
+
+export const DCL_CONTENT_GARBAGE_COLLECTION_TIME = new Histogram({
+  name: 'dcl_content_garbage_collection_time',
+  help: 'Histogram of time spent in garbage collection',
+  labelNames: []
+})
+
+export const DCL_CONTENT_DOWNLOAD_TIME = new Histogram({
```
Comment (agusaldasoro) [inline content/src/ContentMetrics.ts:41]: I'd use a name that reflects that this content download is from the sync, something like `DCL_SYNC_CONTENT_DOWNLOAD_TIME`?

File: content/src/service/synchronization/EventDeployer.ts [hunk 4]
```diff
@@ -35,13 +36,16 @@ export class EventDeployer {
...
+    const stopTimer = DCL_CONTENT_DOWNLOAD_TIME.startTimer({ remote_catalyst: source?.getAddress() || 'undefined' })
     const entityFile: Buffer | undefined = await this.getEntityFile(deployment, source)
+    stopTimer()
 
     const { auditInfo } = deployment
 
     if (entityFile) {
       const isLegacyEntity = !!auditInfo.migrationData
       if (auditInfo.overwrittenBy) {
+        DCL_CONTENT_DOWNLOADED_TOTAL.inc({ overwritten: 'true' })
```
Comment (agusaldasoro) [inline content/src/service/synchronization/EventDeployer.ts:48]: what does this do?

File: content/src/service/synchronization/streaming/EventStreamProcessor.ts [hunk 5]
```diff
@@ -102,7 +109,8 @@ export class EventStreamProcessor {
           }
           done()
         } catch (error) {
-          EventStreamProcessor.LOGGER.debug(
+          DCL_CONTENT_FAILED_DEPLOYMENTS_TOTAL.inc()
```
Comment (agusaldasoro) [inline content/src/service/synchronization/streaming/EventStreamProcessor.ts:112]: Shouldn't this be together with the code that inserts the failed deployments in the database?

File: content/src/service/synchronization/EventDeployer.ts (hunk 4 above)
Comment (menduz) [reply content/src/service/synchronization/EventDeployer.ts:48]: It increments DCL_CONTENT_DOWNLOADED_TOTAL with the tag overwritten: true, the intended usage is to start measuring how many times this branch is taken. and to measure the difference between `DCL_CONTENT_DOWNLOAD_TIME.count <> sum(DCL_CONTENT_DOWNLOADED_TOTAL)` to detect how many entitiyFile are undefined for some reason

File: content/src/service/synchronization/streaming/EventStreamProcessor.ts (hunk 5 above)
Comment (menduz) [reply content/src/service/synchronization/streaming/EventStreamProcessor.ts:112]: The DB of failed entities seems empty, I want to measure how much times this path is being taken without explicitly inserting in the DB

**Comment by agusaldasoro** [comment]
The changes from this PR have been replicated in https://github.com/decentraland/catalyst/pull/647 & https://github.com/decentraland/catalyst/pull/649

**Pull request closed without merging**